client_connection/manager_test.go

```diff
@@ -57,89 +56,129 @@ func (tc *testContext) SetupTest() {
 }
 
 func (tc *testContext) TestWhenNoConnectionIsMadeStatusIsNotConnected() {
-	assert.Equal(tc.T(), ConnectionStatus{NotConnected, "", nil}, tc.connManager.Status())
+	assert.Exactly(tc.T(), statusNotConnected(), tc.connManager.Status())
 }
 
 func (tc *testContext) TestWithUnknownProviderConnectionIsNotMade() {
 	noProposalsError := errors.New("provider has no service proposals")
 
-	assert.Error(tc.T(), tc.connManager.Connect(myID, identity.FromAddress("unknown-node")))
-	assert.Equal(tc.T(), ConnectionStatus{NotConnected, "", noProposalsError}, tc.connManager.Status())
+	assert.Equal(tc.T(), noProposalsError, tc.connManager.Connect(myID, identity.FromAddress("unknown-node")))
+	assert.Equal(tc.T(), statusNotConnected(), tc.connManager.Status())
 
-	assert.False(tc.T(), tc.fakeStatsKeeper.SessionStartMarked)
+	assert.False(tc.T(), tc.fakeStatsKeeper.sessionStartMarked)
 }
 
 func (tc *testContext) TestOnConnectErrorStatusIsNotConnectedAndLastErrorIsSetAndSessionStartIsNotMarked() {
```

Comment:
I don't see where LastErrorIsSet is checked.

Reply:
Good point. Since there is no "last error" anymore, test will be renamed.